### ISSUE TITLE
TypeError: Cannot read property 'map' of undefined

### ISSUE THREAD

**Issue by minh1061998** [basir/node-react-ecommerce#38]
![Untitled](https://user-images.githubusercontent.com/44441160/84219671-40dd9c80-aafb-11ea-8c32-c2bbce23f740.png)
![Untitled1](https://user-images.githubusercontent.com/44441160/84219676-4509ba00-aafb-11ea-8e62-d679a6abbd7b.png)

**Comment by minh1061998** [comment]
Hi basir could you tell me why i have this error pls?

**Comment by vikashchander** [comment]
@minh1061998 
Hi, I think products isn't array so, map method not working

**Comment by prabaljainn** [comment]
yes sir , please help me . I am also stuck here. i have exactly followed your code and nothing is missing. i have checked several times 
even after doing this --
 return loading ? <div>Loading...</div> :
error ? <div>{error}</div> :
 <ul className="products" >
{ 
-------
}

</ul> }

its just not resolving

**Comment by prabaljainn** [comment]
@minh1061998 
brother , My issue is resolved . please check productReducers.js ,  in "  function productListReducer(state= {products: [ ]}, action  )"
you have correctly written "products" and not "product",, it is a very minute mistake took a lot of time to detect.

please let me know, if it was helpfull.
